extensions/ql-vscode/docs/test-plan.md

```diff
@@ -0,0 +1,257 @@
+
+### Test Case 4: Interacting with query history
+
+1. Click a history item (for MRVA):
+   - Check that exporting results works
+   - Check that sorting results works
+   - Check that copying repo lists works 
+2. Open the query directory (containing results):
+   - Check that the correct directory is opened and there are results in it
+3. Open variant analysis on GitHub
+   - Check that the correct workflow is opened
+
+### Test Case 5: Cancelling a variant analysis run
+
+When running one of the above queries, try cancelling one from within VS Code:
+- Check that the query is cancelled and the query history item is updated.
+- Check that the workflow run is also cancelled. 
+- Check that any available results are visible in VS Code.
+
+### Test Case 6: Change to a different colour theme 
+
+When running one of the above queries, try changing to a different colour theme 
+and check that everything looks sensible. Are there any components that are not showing up?
+
+## Optional Test Cases
+
+### 1. Selecting repositories to run on
+
+#### Test case 1: Running a query on a single repository
+1. When the repository exists and is public 
+   1. Has a CodeQL database for the correct language 
+   2. Has a CodeQL database for another language 
+   3. Does not have any CodeQL databases
+2. When the repository exists and is private
+   1. Is accessible and has a CodeQL database
+   2. Is not accessible
+3. When the repository does not exist
+
+#### Test case 2: Running a query on a custom repository list
+1. The repository list is non-empty
+   1. All repositories in the list have a CodeQL database 
+   2. Some but not all repositories in the list have a CodeQL database
+   3. No repositories in the list have a CodeQL database
+2. The repository list is empty
+
+#### Test case 3: Running a query on a built-in repository list (i.e. topN)
+- Check that the query runs and results are shown
+
+#### Test case 4: Running a query on all repositories in an organization
+1. The org exists
+  1. The org contains repositories that have CodeQL databases
+  2. The org contains repositories of the right language but without CodeQL databases
+  3. The org contains repositories not of the right language
+  4. The org contains private repositories	that are inaccessible
+2. The org does not exist
+
+### 2. Using different types of controller repos
+1. The controller is public 
+  1. Can run queries on public repositories
+  2. Can not run queries on private repositories
+2. The controller is private
+  1. Can run queries on public repositories
+  2. Can run queries on private repositories
+3. The controller repo exists but you do not have write access
+4. The controller repo doesn’t exist
+5. The “config field” for the controller repo is not set
+
+### 3. Query History
+1. Query history is loaded when vscode starts
+- Handles when action workflow was cancelled while vscode was closed
+- Starts monitoring variant analyses
+  - When variant analysis state is “pending”
```

Comment:
Very minor nit: These quotes are not the "normal" double quotes, but have some styling applied to them. To make these easier to update, I think it might be easier to replace them by `"`.